_posts/2021-04-09-tests-gives-you-confidence.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: post
+title: "Tests gives you confidence - But data gives you power"
+description: ""
+category:
+tags: [python, pycon.il, elasticsearch, pytest]
+---
+{% include JB/setup %}
+
+# Tests gives you confidence - But data gives you power
+
+## Who Am I
+
+* Israel Fruchter, I'm working in testing for the past 15 year. previusly on the television and home entertiment domain, and nowdays at scylladb.
+
+## What's broken about test reporting
+
+As we all know, most of the testing frameworks we use and love, were written with the intention of running those quick and very precise unit test
+
+But then reality come knocking, products evolves, teams expend, microservices spawns. And you find yourself with silly amount of those functional or integration or selenium based tests, with run runtime of 30secs to 1 hour each, that you are running almost 24/7. Lots information get spit at back you, and you want to collect it put some order into it. see it clearly.
+
+So you look for the defacto standard of reporting from unittest runners, and bam, junit X M L.
+
+You feed it into your CI system, assuming we are all set to go.
+
+(Oh boy someone was naive...)
+
+* In lots CI systems, test report are not exactly first class citizen (if exist at all)
+
+* Test runner have the tendecy to generate the junit.xml only when the whole test session is finished (the horror finding out the 12 hours timeout to you jenkins job, results)
+
+[facepalm gif]
+
+## pytest-elk-reporter enters the stage
+
+So in most case it becomes clear you want to throw your test information into some database.
```

Comment:
give an example of how/why it's useful

Reply:
LOL, me and Roy were arguing on this exact thing for two hours last week. Maybe we should continue fight on the topic so I'll have better examples